streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -272,6 +273,7 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated, f
                 try {
                     Thread.sleep(backoffTimeMs);
                     backoffTimeMs <<= 1;
+                    backoffTimeMs = Math.min(backoffTimeMs, maxBackoffTimeMs);
```

Comment:
Oh. My bad. That is unrelated and I included it to not forget about it -- currently, with exponential increase of waiting time users did report unsatisfying behavior for this. I think we need to fix it... This should also be back ported to `0.10.2`... But maybe it should not be part of this PR. But not sure how else we can get it into `0.11` ;)